galsim/compound.py

```diff
@@ -734,23 +768,32 @@ def __init__(self, obj, real_space=None, gsparams=None):
         sbp = galsim._galsim.SBAutoCorrelate(obj.SBProfile, real_space, gsparams)
         galsim.GSObject.__init__(self, sbp)
 
-        if hasattr(obj,'noise'):
+        if hasattr(obj, 'noise'):
             import warnings
             warnings.warn("Unable to propagate noise in galsim.AutoCorrelation")
 
     @property
     def orig_obj(self): return self._orig_obj
+
     @property
     def real_space(self): return self._real_space
 
+    def __eq__(self, other):
+        return (isinstance(other, galsim.AutoCorrelation) and
+                self.orig_obj == other.orig_obj and
+                self.real_space == other.real_space and
+                self._gsparams == other._gsparams)
+
+    def __hash__(self):
+        return hash(("galsim.AutoCorrelation", self.orig_obj, self.real_space, self._gsparams))
+
     def __repr__(self):
-        return 'galsim.AutoCorrelation(%r, real_space=%r, gsparams=%r)'%(
-                self.orig_obj, self.real_space, self._gsparams)
+        return 'galsim.AutoCorrelation(%r, real_space=%r, gsparams=%r)' % (
+            self.orig_obj, self.real_space, self._gsparams)
 
     def __str__(self):
-        s = 'galsim.AutoCorrelate(%s'%self.orig_obj
-        if self.real_space:
-            s += ', real_space=True'
+        s = 'galsim.AutoCorrelate(%s' % self.orig_obj
+        if self.real_space: s += ', real_space=True'
```

Comment:
This on the other hand, is not a place where the one liner is clearer.  This one should be two lines.  Mostly, you should just trust the original author's choice. That's most often going to be a good, readable style.  In this case, this was originally two lines, and IMO it should stay that way.